src/quark.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use translate::*;
+
+#[repr(C)]
+pub struct Quark(ffi::GQuark);
+
+impl Quark {
+    pub fn from_string(s: &str) -> Quark {
+        unsafe {
+            from_glib(ffi::g_quark_from_string(s.to_glib_none().0))
+        }
+    }
+
+    pub fn from_static_string(s: &'static str) -> Quark {
+        unsafe {
+            from_glib(ffi::g_quark_from_static_string(s.to_glib_none().0))
+        }
+    }
+
+    pub fn to_string(&self) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::g_quark_to_string(self.to_glib()))
+        }
+    }
+
+    pub fn try_string(s: &str) -> Option<Quark> {
+        unsafe {
+            match ffi::g_quark_try_string(s.to_glib_none().0) {
+                0 => None,
+                x => Some(from_glib(x)),
+            }
+        }
+    }
+
+    pub fn intern_string(s: &str) -> Option<String> {
+        unsafe {
+            from_glib_none(ffi::g_intern_string(s.to_glib_none().0))
+        }
+    }
+
+    pub fn intern_static_string(s: &str) -> Option<String> {
```

Comment:
I'm just worry about this implementation: do we have ownership on the returned string or not? If so, we'll have issues because we're not supposed to free it...

Reply:
`from_glib_none` IMHO is right, but this function don't make sense in rust and better not implemented.
Also it already in https://github.com/gtk-rs/glib/blob/master/src/auto/functions.rs#L614 and better ignored with `intern_string`

Reply:
Sounds good to me!

Reply:
The returned string here is the same as below in to_string(). Static string slice would be correct

But you can't bind the static variant, only the other. And both intern functions are not very useful in Rust